File: ruby/quality_rating.rb
class QualityRating

  MAX_QUALITY = 50
  MIN_QUALITY = 0

  def initialize(quality)
    @quality = quality
  end

  def modify(value_to_add)
    if value_to_add.zero?
      return
    end

    new_quality = @quality + value_to_add
    @quality =
      if new_quality > MAX_QUALITY
        MAX_QUALITY
      elsif new_quality < MIN_QUALITY
        MIN_QUALITY
      else
        new_quality
      end
  end

  def get
    @quality
  end

  def set(new_quality)
    @quality = new_quality
  end
end

File: cpp-catch2/src/QualityRating.cpp
//
// Created by markd on 3/9/2021.
//

#include "QualityRating.h"

QualityRating::QualityRating(int initialQuality) {
    quality = initialQuality;
}

void QualityRating::modify(int valueToAdd) {
    if (valueToAdd == 0){
        return;
    }
    int newQuality = quality + valueToAdd;
    if(newQuality > MAX_QUALITY){
        quality = MAX_QUALITY;
    }
    else if(newQuality < MIN_QUALITY){
        quality = MIN_QUALITY;
    }
    else {
        quality = newQuality;
    }
}

int QualityRating::get() {
    return quality;
}

void QualityRating::set(int newQuality) {
    quality = newQuality;
}
